### PR TITLE
fix: dedupe ContentRule records by name and enforce uniqueness

### LINKED ISSUE #1530: Rule name cleanup
We have a lot of duplicate names for Rules (often as a result of input errors), which could potentially cause confusion when players are making custom content. In some instances, there are different rules that have the same name. However, because we don’t actually host rules, merging these shouldn’t matter.

One consideration is that this might cause issues with page references, I think solution would just be that we sometimes give multiple references for the same rule and hope that players can figure it out.

### PR DESCRIPTION
**Pull request by tgvashworth**
## Summary

- Adds data migration `0167_dedupe_content_rules` that merges duplicate-name `ContentRule` records: picks the canonical record per name by reference-count desc (with oldest `created` as tiebreaker), repoints every reference (`ContentFighter.rules`, `ListFighter.custom_rules`/`disabled_rules`, `ContentModFighterRule.rule`, and `CustomContentPackItem` GFK rows), then deletes the duplicates.
- Adds migration `0168_contentrule_name_unique` setting `unique=True` on `ContentRule.name` to prevent regressions.
- Page references match rules by title string (not FK), so merging by name preserves all `ContentPageRef` lookups.

## Prod audit informing the approach

- 436 total rules → 14 duplicate-name groups, 30 records involved.
- All duplicates have empty `description` and zero `CustomContentPackItem` membership, so no text-merge or pack-conflict edge cases in practice.
- Case-insensitive matches don't add any groups beyond exact-name matches.

## Test plan

- [x] Local DB forked from `gyrinx_main` template: pre = 404 rules / 13 dupe groups → post = 390 rules / 0 dupe groups. All references preserved on canonical records (snapshot-and-compare).
- [x] `IntegrityError` raised when attempting to create a duplicate `ContentRule` after migration.
- [x] Full pytest suite: 2596 passed, 13 skipped — no regressions from the new constraint.
- [ ] Confirm migration runs cleanly in staging/prod before merging (no rollback path — `reverse_code` is a no-op).

## Notes

- The migration is irreversible — `reverse_code` cannot reconstruct deleted rule rows. Documented in the module docstring.
- Historical (`django-simple-history`) rows for deleted duplicates remain pointing at the old rule_id. Acceptable since history is forensic, not authoritative.
- Pack-item GFK path is defensive: if a duplicate is in a pack that already contains the canonical, the duplicate's pack item is deleted rather than updated, to avoid colliding with the conditional unique constraint on `CustomContentPackItem`.

Closes #1530

### CODE REVIEW

File: gyrinx/content/migrations/0168_dedupe_content_rules.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    CustomContentPackItem = apps.get_model("core", "CustomContentPackItem")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    rule_ct = ContentType.objects.get(app_label="content", model="contentrule")
+
+    dupe_groups = list(
+        ContentRule.objects.values("name")
+        .annotate(c=Count("id"))
+        .filter(c__gt=1)
+        .order_by("name")
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/content/migrations/0168_dedupe_content_rules.py:33]: Django migrations use the historical model with a plain `Manager`, not the model's custom `ContentManager` — `apps.get_model()` strips custom managers and `Content` has no `base_manager_name`. Verified locally: live `ContentRule.objects` is `ContentManagerFromContentQuerySet` (390 visible) vs. migration `apps.get_model("content", "ContentRule").objects` is `Manager` (398 visible — sees the 8 pack-scoped rules). Post-migration referential-integrity scan: 0 dangling refs in `content_contentfighter_rules`, `core_listfighter_custom_rules`, `core_listfighter_disabled_rules`, `content_contentmodfighterrule.rule_id`, or `CustomContentPackItem`. The audit-finding numbers in the PR description came from prodshell (live manager), but the migration runs comprehensively.

File: gyrinx/content/migrations/0168_dedupe_content_rules.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    def ref_count(rule):
+        return (
+            ContentFighter.objects.filter(rules=rule).count()
+            + ListFighter.objects.filter(custom_rules=rule).count()
+            + ListFighter.objects.filter(disabled_rules=rule).count()
+            + ContentModFighterRule.objects.filter(rule=rule).count()
+            + CustomContentPackItem.objects.filter(
+                content_type=rule_ct, object_id=rule.id
+            ).count()
+        )
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/content/migrations/0168_dedupe_content_rules.py:49]: Django migrations use the historical model with a plain `Manager`, not the model's custom `ContentManager` — `apps.get_model()` strips custom managers and `Content` has no `base_manager_name`. Verified locally: live `ContentRule.objects` is `ContentManagerFromContentQuerySet` (390 visible) vs. migration `apps.get_model("content", "ContentRule").objects` is `Manager` (398 visible — sees the 8 pack-scoped rules). Post-migration referential-integrity scan: 0 dangling refs in `content_contentfighter_rules`, `core_listfighter_custom_rules`, `core_listfighter_disabled_rules`, `content_contentmodfighterrule.rule_id`, or `CustomContentPackItem`. The audit-finding numbers in the PR description came from prodshell (live manager), but the migration runs comprehensively.

File: gyrinx/content/migrations/0168_dedupe_content_rules.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                cf.rules.add(canonical)
+                cf.rules.remove(dup)
+
+            for lf in ListFighter.objects.filter(custom_rules=dup):
+                lf.custom_rules.add(canonical)
+                lf.custom_rules.remove(dup)
+
+            for lf in ListFighter.objects.filter(disabled_rules=dup):
+                lf.disabled_rules.add(canonical)
+                lf.disabled_rules.remove(dup)
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/content/migrations/0168_dedupe_content_rules.py:78]: Django migrations use the historical model with a plain `Manager`, not the model's custom `ContentManager` — `apps.get_model()` strips custom managers and `Content` has no `base_manager_name`. Verified locally: live `ContentRule.objects` is `ContentManagerFromContentQuerySet` (390 visible) vs. migration `apps.get_model("content", "ContentRule").objects` is `Manager` (398 visible — sees the 8 pack-scoped rules). Post-migration referential-integrity scan: 0 dangling refs in `content_contentfighter_rules`, `core_listfighter_custom_rules`, `core_listfighter_disabled_rules`, `content_contentmodfighterrule.rule_id`, or `CustomContentPackItem`. The audit-finding numbers in the PR description came from prodshell (live manager), but the migration runs comprehensively.

File: gyrinx/content/migrations/0168_dedupe_content_rules.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+            ContentModFighterRule.objects.filter(rule=dup).update(rule=canonical)
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply gyrinx/content/migrations/0168_dedupe_content_rules.py:80]: Django migrations use the historical model with a plain `Manager`, not the model's custom `ContentManager` — `apps.get_model()` strips custom managers and `Content` has no `base_manager_name`. Verified locally: live `ContentRule.objects` is `ContentManagerFromContentQuerySet` (390 visible) vs. migration `apps.get_model("content", "ContentRule").objects` is `Manager` (398 visible — sees the 8 pack-scoped rules). Post-migration referential-integrity scan: 0 dangling refs in `content_contentfighter_rules`, `core_listfighter_custom_rules`, `core_listfighter_disabled_rules`, `content_contentmodfighterrule.rule_id`, or `CustomContentPackItem`. The audit-finding numbers in the PR description came from prodshell (live manager), but the migration runs comprehensively.
